### PR TITLE
[SandboxIR] Add bazel support

### PR DESCRIPTION
**Pull request by Sterling-Augustine**

### CODE REVIEW

File: utils/bazel/llvm-project-overlay/llvm/BUILD.bazel [hunk 1]
```diff
@@ -1490,6 +1504,22 @@ cc_library(
     ],
 )
 
+cc_library(
+    name = "SandboxVectorizer",
+    srcs = glob([
+        "lib/Transforms/Vectorize/SandboxVectorizer/*.cpp",
```
Comment (aeubanks) [inline utils/bazel/llvm-project-overlay/llvm/BUILD.bazel:1510]: for consistency with the CMake build, this should be part of `Vectorize` above

File: utils/bazel/llvm-project-overlay/llvm/BUILD.bazel [hunk 2]
```diff
@@ -1446,6 +1446,20 @@ cc_library(
     ],
 )
 
+cc_library(
+    name = "SandboxIR",
+    srcs = glob([
+        "lib/SandboxIR/*.cpp",
+    ]),
+    hdrs = glob(["include/llvm/SandboxIR/*.h"]) +
+    ["include/llvm/SandboxIR/SandboxIRValues.def"],
```
Comment (slackito) [inline utils/bazel/llvm-project-overlay/llvm/BUILD.bazel:1455]: The .def file is just a bunch of X macros so it should probably go in `textual_hdrs`

https://bazel.build/reference/be/c-cpp#cc_library.textual_hdrs:
> This is the location for declaring header files that cannot be compiled on their own; that is, they always need to be textually included by other source files to build valid code.

File: utils/bazel/llvm-project-overlay/llvm/BUILD.bazel (hunk 1 above)
Comment (Sterling-Augustine) [reply utils/bazel/llvm-project-overlay/llvm/BUILD.bazel]: Done

File: utils/bazel/llvm-project-overlay/llvm/BUILD.bazel (hunk 2 above)
Comment (Sterling-Augustine) [reply utils/bazel/llvm-project-overlay/llvm/BUILD.bazel]: Done

**Review by slackito — APPROVED** [review]

**Comment by Sterling-Augustine** [comment]
Addressed comments.

**Pull request merged**